test/Webserver_test.cc

```diff
@@ -1,48 +1,45 @@
 #include "gtest/gtest.h"
 #include "gmock/gmock.h"
 #include "Webserver.h"
+#include "request_handler.h"
 #include "config_parser.h"
-#include "HttpRequest.h"
-#include "HttpResponse.h"
-#include "HttpConstants.h"
 #include <fstream>
 #include <iostream>
 #include <cstdio>
 
 using ::testing::Return;
 using ::testing::_;
 
-class MockHttpRequest : public HttpRequest {
-public:
-    MOCK_METHOD0(to_string, std::string());
-    MOCK_METHOD0(get_file, std::string());
-    MOCK_METHOD0(get_type, std::string());
-    MOCK_METHOD1(create_request, int(boost::asio::streambuf& buffer));
-};
+// class MockHttpRequest : public HttpRequest {
```

Comment:
yea i guess we should delete it. making a mock request and calling request handler be for unit test for each handler. For compilation it doesn't matter but it might be faster it take it out.